atc/gc/version_reaper.go

```diff
@@ -0,0 +1,42 @@
+package gc
+
+import (
+	"context"
+	"github.com/concourse/concourse/atc/db"
+	"time"
+
+	"code.cloudfoundry.org/lager/lagerctx"
+	"github.com/concourse/concourse/atc/metric"
+)
+
+type versionReaper struct {
+	conn                   db.Conn
+	maxVersionsPerResource int
+}
+
+func NewVersionReaper(conn db.Conn, maxVersionsPerResource int) *versionReaper {
+	if maxVersionsPerResource < 100 {
```

Comment:
Was there a reason why we want to have a minimum of 100 versions per resource?

Reply:
This is just to prevent from unreasonable min value, say, 1 version to retain.

Reply:
I'm not sure how I feel about having a hardcoded strict minimum because there could be the case where someone only cares about having fewer than 100 versions per resource? And when they try to set less than 100 it is not clear why their set minimum isn't respected (if we do want to have a strict min maybe that should be a flag validation)? Is there any benefit to setting this min value?

Reply:
@clarafu  My thought is that we should prevent user from setting some unreasonable value, for example 1. If a user sets only to retain 1 version per resource, I am afraid Concourse not behavior properly. I agree 100 is a random coming out from my brain, I just feel it should be a reasonable limitation. If you believe 1 is a safe limitation, then I can remove this check.

Reply:
If someone sets it to be a really low value like 1, I don't think there will be necessarily any concourse problems that come about from it. There might be things that will incentivize users from not setting it to such a low value, such as a version not being able to run through the entire pipeline in time before another version comes in and ends up deleting the previous version? Maybe we can be clear in our documentation that if a low value is set, it can possibly result in your versions not making it completely through your pipelines because it gets deleted before it can do so.

Reply:
Okey, I just deleted the check. Please see if you have more comments.